### PR TITLE
Fix tie handling when sorting strides in JIT TensorType

### PR DESCRIPTION
**Pull request by kevinstephano**
When Pytorch permutes a Tensor it alters the strides (meta data) instead of rearranging the data.  There is an assumption in Pytorch that you can re-determine the original order of the indices (that are contiguous) from sorting the strides. Therefore, they don't carry any ordering information. If you don't have any indices with a dimension size of 1, this works fine.  If you do have indices with dimension size of 1, like you find with broadcast operations, you inject ties into this sorting.  If you have 3 or more ties, you don't have enough information to exactly get back the original ordering of the indices with the information you have.  One can say this doesn't matter because the order of 3 indices of size 1 doesn't impact your data.  However,  the JIT calculates its output Tensor dimensions and order by "merging" its input Tensors properties based on the index order.  If the input Tensors index order doesn't match, you can inadvertently permute your output.
 
The JIT Runtime code that calculates the stride properties (index ordering to be contiguous) has a bug in it when there are two indices that have a stride size tie. The comparator used for sorting made a bad assumption about arbitrating the tie based on index order.  I fixed that issue by changing to arbitrate on whether the index was of size 1.  For ascending order strides, I wanted a dimension size of size 1 to be "smaller" as it creates the condition of the stride tie.

### CODE REVIEW

File: aten/src/ATen/core/type.cpp [hunk 1]
```diff
@@ -870,14 +870,29 @@ VaryingShape<Stride> TensorType::computeStrideProps(
   std::vector<size_t> stride_indices(sizes.size());
   std::iota(stride_indices.begin(), stride_indices.end(), 0);
 
+  // Sorting strides in ascending order
+  // Warning: With more than 2 strides of the same size
```
Comment (jjsjann123) [inline aten/src/ATen/core/type.cpp]: nitpick: `2 strides of the same size` is confusing. IIUC, we just can't handle more than 1 dimension with size == 1, this doesn't seem to be saying that.

Comment (kevinstephano) [reply aten/src/ATen/core/type.cpp]: It made perfect sense to me in my mind when writing it for myself. :-D

File: aten/src/ATen/core/type.cpp [hunk 2]
```diff
@@ -870,14 +870,29 @@ VaryingShape<Stride> TensorType::computeStrideProps(
...
   std::sort(
       stride_indices.begin(),
       stride_indices.end(),
-      [&strides](const int& a, const int& b) {
-        // break ties in case of unsqueezed dims
-        // i.e. (1, 1, 5)
+      [&strides, &sizes](const int& a, const int& b) {
         if (strides[a] == strides[b]) {
-          return a > b;
+          return sizes[a] == 1;
```
Comment (jjsjann123) [inline aten/src/ATen/core/type.cpp]: nitpick: let's play on the safe side. In the ambiguous case, let's default NCHW.

i.e. `sizes[a] == 1 if sizes[b] != sizes[a] else a > b`

**Review by jjsjann123 — APPROVED** [review]
LGTM

**Pull request merged**